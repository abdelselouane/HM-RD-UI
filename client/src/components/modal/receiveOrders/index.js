export { default as InfoModal } from './infoModal';