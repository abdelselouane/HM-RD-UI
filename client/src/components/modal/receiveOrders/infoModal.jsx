import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Table from '../../table';
import Modal from '../modal';

const ScrollableOverflowTableContent = styled.div`
  .scrollable-table {
    overflow-y: hidden !important;
  }
`;

const InfoModal = ({ onPrimaryClick, onSecondaryClick, rows, columns, selectedPurchaseOrderNbr, rowLegnth = rows.length }) => (
  <Modal {...{
    styles: {
      width: '50%'
    },
    scrollable: (rowLegnth > 4) ? true : false,
    title: {
      className: 'info',
      text: `PO #${selectedPurchaseOrderNbr} Please select the quantity you would like to receive!`,
      icon: 'information-outlined',
      closeModal: () => onSecondaryClick()
    },
    content: {
      className: '',
      text: '',
      body: (
        <ScrollableOverflowTableContent className="body">
          <Table
            rows={rows}
            columns={columns}
            showResults={!!rows.length}
            isAlternating={false}
          />
        </ScrollableOverflowTableContent>)
    },
    buttons: [{
      text: 'Cancel',
      className: 'md',
      onClick: () => onSecondaryClick()
    },
    {
      text: 'Receive Part',
      className: 'primary md',
      onClick: () => onPrimaryClick()
    }]
  }} />
);
export default InfoModal;