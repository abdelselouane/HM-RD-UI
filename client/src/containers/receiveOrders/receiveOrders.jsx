/* globals document */
import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Counter from '../../components/counter';
import Table from '../../components/table';
import { InfoModal } from '../../components/modal';
import * as PoActions from '../../actions/purchaseOrdersActions';
import * as NotificationActions from '../../actions/notificationActions';
import { timingSafeEqual } from 'crypto';

export class ReceiveOrders extends React.Component {
  state = {
    selectedParts: [],
    showReceivePartModal: false,
    selectedPurchaseOrderNbr: '',
    sortedColumn: 'orderCreatedDate'
  };

  componentWillMount() {
    this.props.actions.getPurchaseOrders();
  }

  columns = [
    {
      name: 'Date Created',
      key: 'orderCreatedDate',
      sortable: true,
      orderBy: {
        key: 'orderCreatedDate',
        className: ''
      }
    }, {
      name: 'Purchase Order #',
      key: 'purchaseOrderNbr',
      sortable: true,
      orderBy: {
        key: 'purchaseOrderNbr',
        className: ''
      }
    }, {
      name: 'Work order #',
      key: 'workOrderNbr',
      sortable: true,
      orderBy: {
        key: 'workOrderNbr',
        className: ''
      }
    }, {
      name: 'Status',
      key: 'purchaseOrderStatDesc',
      sortable: true,
      orderBy: {
        key: 'purchaseOrderStatDesc',
        className: ''
      }
    }, {
      name: 'Receive',
      key: 'receive'
    }
  ];

  partsDetailsColumns = [
    {
      name: 'Part Number',
      key: 'partNbr'
    }, {
      name: 'Description',
      key: 'partDescription'
    }, {
      name: 'Location',
      key: 'location'
    }, {
      name: 'Received Quantity',
      key: 'receivedQuantity'
    }, {
      name: 'Receiving',
      key: 'receive'
    }
  ];

  getSortedOrders = orders => {
    if (orders && orders.length > 0) {
      let sort = '';
      const column = this.state.sortedColumn;
      this.columns.map(col => {
        if (col.key === column) {
          if (col.orderBy.className === '' || col.orderBy.className === 'descending') {
            sort = 'ascending';
          } else {
            sort = 'descending';
          }
        }
      });

      orders.sort((a, b) => {
        if (sort === 'ascending') {
          const nameA = a[column].toUpperCase(); // ignore upper and lowercase
          const nameB = b[column].toUpperCase(); // ignore upper and lowercase

          if (nameA < nameB) {
            return -1;
          }
          if (nameA > nameB) {
            return 1;
          }

          return 0;
        }
        const nameA = a[column].toUpperCase(); // ignore upper and lowercase
        const nameB = b[column].toUpperCase(); // ignore upper and lowercase

        if (nameA > nameB) {
          return -1;
        }
        if (nameA < nameB) {
          return 1;
        }
        return 0;
      });
    }
    return orders;
  };

  handleSortClick = column => {
    this.columns.map(col => {
      if (col.key === column) {
        if (col.orderBy.className === '' || col.orderBy.className === 'descending') {
          col.orderBy.className = 'ascending';
        } else {
          col.orderBy.className = 'descending';
        }
      }
    });
    this.setState({
      sortedColumn: column
    });
  };

  handleCounterChange = (name, value) => {
    const { selectedParts } = this.state;
    selectedParts.map(part => {
      if (part.partNbr === name) {
        part.currentQuantityReceived = value;
      }
      if (part.currentQuantityReceived > 0) {
        part.invalidQuantity = false;
      }
    });
    this.setState({
      selectedParts
    });
  }

  handleLocationChange = (partNbr, event) => {
    const { selectedParts } = this.state;
    selectedParts.map(part => {
      if (part.partNbr === partNbr) {
        part.partLocation = event.target.value;
        return part;
      }
    });
    this.setState({
      selectedParts
    });
  }

  handleReceivePart = (purchaseOrderNbr) => {
    const { orders } = this.props;

    const selectedOrder = orders.filter((order) => purchaseOrderNbr === order.purchaseOrderNbr);

    const selectedParts = selectedOrder[0].partsDetails.map(part => ({
      ...part,
      invalidQuantity: false
    }));

    this.setState({
      selectedPurchaseOrderNbr: purchaseOrderNbr,
      selectedParts,
      showReceivePartModal: true
    });
  }

  handleCloseModal = () => {
    this.setState({
      showReceivePartModal: false,
      selectedParts: [],
      selectedPurchaseOrderNbr: ''
    });
  }

  handleSaveReceivePart = () => {
    const { selectedParts } = this.state;
    const validParts = [];
    selectedParts.map(part => {
      if (part.currentQuantityReceived > 0) {
        validParts.push(part);
      }
    });
    this.setState({
      selectedParts
    });

    if (validParts.length > 0) {
      const { selectedPurchaseOrderNbr } = this.state;
      const { orders } = this.props;
      const selectedOrder = orders.filter((order) => selectedPurchaseOrderNbr === order.purchaseOrderNbr);

      const payload = {
        locationNbr: selectedOrder[0].locationNbr,
        partsDetails: selectedParts.map(part => ({
          brandName: part.brandName,
          brandNbr: part.brandNbr,
          currentQuantityReceived: part.currentQuantityReceived,
          partDescription: part.partDescription,
          partLocation: part.partLocation,
          partNbr: part.partNbr,
          previousQuantityReceived: part.previousQuantityReceived,
          quantityOrdered: part.quantityOrdered
        })),
        partsOrderId: selectedOrder[0].partsOrderId,
        purchaseOrderNbr: selectedOrder[0].purchaseOrderNbr
      };

      this.handleUpdateReceivePartOrder(payload);
    } else {
      selectedParts.map(part => {
        if (part.currentQuantityReceived === 0) {
          part.invalidQuantity = true;
        }
      });
    }
  }

  handleUpdateReceivePartOrder = async (payload) => {
    await this.props.actions.updateReceivePartOrder(payload);
    if (this.props.updatePartOrdersError) {
      this.props.actions.showMessage(
        'System Failure: Unable to receive parts, please try again later'
      );
      setTimeout(() => this.props.actions.hideMessage(), 10000);
    } else {
      this.props.actions.showSuccessMessage(
        'Parts Received: The parts have been received in to your inventory'
      );
      setTimeout(() => this.props.actions.hideMessage(), 10000);
    }
    this.handleCloseModal();
  };

  render() {
    const { orders, inventory } = this.props;
    const {
      showReceivePartModal,
      selectedPurchaseOrderNbr,
      selectedParts
    } = this.state;

    let selectedInvPart = [];
    const partsDetailsRows = selectedParts.map((part) => {
      selectedInvPart = inventory.filter(inventoryPart => part.partNbr === inventoryPart.partNbr);
      let invLocation;
      if (selectedInvPart.length > 0 && selectedInvPart[0].location) {
        invLocation = selectedInvPart[0].location;
      } else {
        invLocation = (
          <div className="text-input-container" >
            <input
              id={`location${part.partNbr}`}
              type="text"
              value={part.partLocation}
              onChange={(event) => { this.handleLocationChange(part.partNbr, event); }}
              placeholder="Not Specified"
              style={{ marginTop: 0 }}
            />
          </div>);
      }
      return ({
        ...part,
        key: part.partNbr,
        location: invLocation,
        receivedQuantity: `${part.previousQuantityReceived} of ${part.quantityOrdered}`,
        receive: <div className="qty-counter counter-input sm" aria-invalid={part.invalidQuantity}>
          <Counter
            name={part.partNbr}
            value={part.currentQuantityReceived}
            min={0}
            max={part.quantityOrdered - part.previousQuantityReceived}
            handleChangeValue={this.handleCounterChange}
          />
        </div>
      });
    });

    const rows = this.getSortedOrders([...orders]).map(({ partDetails, ...order }, index) => ({
      ...order,
      key: order.purchaseOrderNbr,
      receive: <button
        id={`${index}-${order.purchaseOrderNbr}`}
        onClick={() => this.handleReceivePart(order.purchaseOrderNbr)}
        className="button md"
        style={{ margin: '10px 0px' }}
      >Receive Order
      </button>
    }));
    return (
      <React.Fragment>

        <Table
          rows={rows}
          columns={this.columns}
          showResults={!!rows.length}
          isExpandable
          isAlternating={false}
          handleSortClick={this.handleSortClick}
        />
        {showReceivePartModal &&
          ReactDOM.createPortal(
            <InfoModal
              onPrimaryClick={this.handleSaveReceivePart}
              onSecondaryClick={this.handleCloseModal}
              rows={partsDetailsRows}
              columns={this.partsDetailsColumns}
              selectedPurchaseOrderNbr={selectedPurchaseOrderNbr}
            />,
            document.querySelector('#modal')
          )}
      </React.Fragment>
    );
  }
}

ReceiveOrders.propTypes = {
  actions: PropTypes.shape({
    getPurchaseOrders: PropTypes.func.isRequired
  }).isRequired,
  orders: PropTypes.arrayOf(
    PropTypes.shape({
      orderCreatedDate: PropTypes.string,
      workOrderNbr: PropTypes.string,
      purchaseOrderStatDesc: PropTypes.string,
      purchaseOrderNbr: PropTypes.string,
      partsDetails: PropTypes.arrayOf(PropTypes.shape({
        quantityOrdered: PropTypes.number.isRequired,
        previousQuantityReceived: PropTypes.number.isRequired,
        partNbr: PropTypes.string.isRequired,
        partDescription: PropTypes.string.isRequired
      }))
    })
  ).isRequired
};

const mapStateToProps = state => ({
  orders: state.purchaseOrders.orders,
  updatePartOrdersError: state.purchaseOrders.error,
  inventory: state.inventory.items
});
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    ...PoActions,
    ...NotificationActions
  }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ReceiveOrders);
