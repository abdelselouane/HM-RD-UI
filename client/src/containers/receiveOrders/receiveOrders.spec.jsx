/* global describe, beforeAll, it, expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ReceiveOrders } from './receiveOrders';

import Table from '../../components/table';
import { consolidateStreamedStyles } from 'styled-components';

jest.mock('../../actions/purchaseOrdersActions', () => ({
  getPurchaseOrders: () => dispatch => dispatch({ type: 'MOCK' })
}));

describe('<ReceiveOrders />', () => {
  let wrapper;
  let props;
  let mockOrder1;
  let mockOrder2;
  let mockInventory;

  const mockPart1 = {
    partNbr: '310-0101',
    brandNbr: 1001,
    brandName: 'MAKITA',
    partLocation: '',
    quantityOrdered: 1,
    partDescription: 'GASKET, TRANSMISSION',
    partsOrderDetailId: 1,
    previousQuantityReceived: 1,
    currentQuantityReceived: 0
  };

  const mockPart2 = {
    partNbr: '310-0102',
    brandNbr: 1001,
    brandName: 'MAKITA',
    partLocation: '',
    quantityOrdered: 10,
    partDescription: 'GASKET, TRANSMISSION-2',
    partsOrderDetailId: 2,
    previousQuantityReceived: 1,
    currentQuantityReceived: 0
  };

  const payLoad = {
    locationNbr: '9710',
    partsDetails:
      [{
        brandName: 'MAKITA',
        brandNbr: 1001,
        currentQuantityReceived: 1,
        partDescription: 'GASKET, TRANSMISSION-2',
        partLocation: '',
        partNbr: '310-0103',
        previousQuantityReceived: 1,
        quantityOrdered: 10
      }],
    partsOrderId: 1,
    purchaseOrderNbr: '971072952'
  };

  const updatedPayLoad = {
    locationNbr: '9710',
    partsDetails:
      [{
        partNbr: '310-0101',
        brandNbr: 1001,
        brandName: 'MAKITA',
        partLocation: '',
        quantityOrdered: 1,
        partDescription: 'GASKET, TRANSMISSION',
        previousQuantityReceived: 1,
        currentQuantityReceived: 0
      },
      {
        brandName: 'MAKITA',
        brandNbr: 1001,
        currentQuantityReceived: 1,
        partDescription: 'GASKET, TRANSMISSION-2',
        partLocation: '',
        partNbr: '310-0103',
        previousQuantityReceived: 1,
        quantityOrdered: 10
      }
      ],
    partsOrderId: 1,
    purchaseOrderNbr: '971072952'
  };

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn();
    mockOrder1 = {
      locationNbr: '9710',
      orderCreatedDate: '2018-06-04',
      workOrderNbr: 'mock#',
      purchaseOrderStatDesc: 'SUBMITTED',
      purchaseOrderNbr: '971072952',
      partsOrderId: 1,
      partsDetails: [mockPart1, mockPart2]
    };
    mockOrder2 = {
      locationNbr: '9710',
      orderCreatedDate: '2018-06-04',
      workOrderNbr: 'mock#',
      purchaseOrderStatDesc: 'SUBMITTED',
      purchaseOrderNbr: '871072952',
      partsOrderId: 1,
      partsDetails: [mockPart1, mockPart2]
    };

    mockInventory = {
      allocatedQuantity: 0,
      availableQuantity: 10,
      brandName: 'Eddy Floor Scraper',
      brandNbr: 1028,
      location: 'A4',
      partDescription: 'GASKET, TRANSMISSION',
      partNbr: '310-0101',
      storeNbr: '9710'
    };

    props = {
      actions: {
        getPurchaseOrders: async () => ({ payload: [mockOrder1] }),
        updateReceivePartOrder: jest.fn(),
        showSuccessMessage: jest.fn(),
        showMessage: jest.fn()
      },
      orders: [mockOrder1],
      inventory: [mockInventory],
      updatePartOrdersError: false
    };


    wrapper = shallow(<ReceiveOrders {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render page with open order table', () => {
    expect(wrapper.find(Table)).toHaveLength(1);
  });

  it('should call handleReceivePart when Receive Part button is clicked', () => {
    const wrapperInstance = wrapper.instance();
    wrapper.find(Table).shallow().find('.receive-col').first()
      .simulate('click');
    wrapperInstance.handleReceivePart('971072952');
    const mockPart1Ar = [mockPart1, mockPart2];
    expect(wrapperInstance.state.selectedPurchaseOrderNbr).toBe('971072952');
    expect(wrapperInstance.state.selectedParts).toMatchObject(mockPart1Ar);
    expect(wrapperInstance.state.showReceivePartModal).toBe(true);
  });

  it('should open modal when receive order button is clicked', () => {
    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleReceivePart('971072952');
    expect(wrapperInstance.state.showReceivePartModal).toBe(true);
  });

  it('should close modal when handleCloseModal is called', () => {
    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleCloseModal();
    expect(wrapperInstance.state.showReceivePartModal).toBe(false);
  });

  it('should set the currentQuantityReceived on handleCounterChange', () => {
    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleReceivePart('971072952');
    expect(wrapperInstance.state.showReceivePartModal).toBe(true);
    wrapperInstance.handleCounterChange('310-0102', 5);
    expect(wrapperInstance.state.selectedParts[0].currentQuantityReceived).toEqual(0);
    expect(wrapperInstance.state.selectedParts[1].currentQuantityReceived).toEqual(5);
  });

  it('should set the sortedColumn on handleSortClick', () => {
    const column = 'purchaseOrderNbr';
    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleSortClick(column);
    expect(wrapperInstance.columns[1].orderBy.className).toEqual('ascending');
    expect(wrapperInstance.state.sortedColumn).toEqual(column);
    wrapperInstance.handleSortClick(column);
    expect(wrapperInstance.state.sortedColumn).toEqual(column);
    expect(wrapperInstance.columns[1].orderBy.className).toEqual('descending');
  });

  it('should sort the orders on getSortedOrders with orderCreatedDate', () => {
    const wrapperInstance = wrapper.instance();
    const order1 = {
      locationNbr: '9710',
      orderCreatedDate: '2018-06-04',
      workOrderNbr: 'mock#',
      purchaseOrderStatDesc: 'SUBMITTED',
      purchaseOrderNbr: '971072952',
      partsOrderId: 1,
      partsDetails: [mockPart1, mockPart2]
    };
    const order2 = {
      locationNbr: '9710',
      orderCreatedDate: '2018-06-04',
      workOrderNbr: 'mock#',
      purchaseOrderStatDesc: 'SUBMITTED',
      purchaseOrderNbr: '871072952',
      partsOrderId: 1,
      partsDetails: [mockPart1, mockPart2]
    };
    const order3 = {
      locationNbr: '9710',
      orderCreatedDate: '2017-06-04',
      workOrderNbr: 'mock#',
      purchaseOrderStatDesc: 'SUBMITTED',
      purchaseOrderNbr: '971072952',
      partsOrderId: 1,
      partsDetails: [mockPart1, mockPart2]
    };

    wrapperInstance.setState({ sortedColumn: 'orderCreatedDate' });
    wrapperInstance.columns[0].orderBy.className = 'descending';
    const sortedOrders1 = wrapperInstance.getSortedOrders([order1, order2]);
    expect(sortedOrders1[0]).toMatchObject(order1);

    wrapperInstance.columns[0].orderBy.className = 'descending';
    const sortedOrders2 = wrapperInstance.getSortedOrders([order1, order3]);
    expect(sortedOrders2[0]).toMatchObject(order3);

    wrapperInstance.columns[0].orderBy.className = 'descending';
    const sortedOrders3 = wrapperInstance.getSortedOrders([order3, order1]);
    expect(sortedOrders3[0]).toMatchObject(order3);

    wrapperInstance.columns[0].orderBy.className = 'ascending';
    const sortedOrders4 = wrapperInstance.getSortedOrders([order1, order2]);
    expect(sortedOrders4[0]).toMatchObject(order1);

    wrapperInstance.columns[0].orderBy.className = 'ascending';
    const sortedOrders5 = wrapperInstance.getSortedOrders([order1, order3]);
    expect(sortedOrders5[0]).toMatchObject(order1);

    wrapperInstance.columns[0].orderBy.className = 'ascending';
    const sortedOrders6 = wrapperInstance.getSortedOrders([order3, order1]);
    expect(sortedOrders6[0]).toMatchObject(order1);
  });

  it('should handle partLocation change on handleCounterChange', () => {
    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleReceivePart('971072952');
    expect(wrapperInstance.state.showReceivePartModal).toBe(true);
    const mockEvent = {
      target: {
        value: 'A12',
      }
    };
    wrapperInstance.handleLocationChange('310-0101', mockEvent);
    expect(wrapperInstance.state.selectedParts[0].partLocation).toEqual('A12');
  });

  it('should handle save receive part', () => {
    const mockHandleUpdateReceivePartOrder = jest.fn();

    const mockPart = {
      partNbr: '310-0103',
      brandNbr: 1001,
      brandName: 'MAKITA',
      partLocation: '',
      quantityOrdered: 10,
      partDescription: 'GASKET, TRANSMISSION-2',
      partsOrderDetailId: 2,
      previousQuantityReceived: 1,
      currentQuantityReceived: 1
    };

    const mockPart1Ar = [mockPart1, mockPart];

    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleUpdateReceivePartOrder = mockHandleUpdateReceivePartOrder;
    wrapper.instance().forceUpdate();
    wrapperInstance.state.selectedParts = mockPart1Ar;
    wrapperInstance.state.selectedPurchaseOrderNbr = '971072952';
    wrapperInstance.handleSaveReceivePart();
    expect(wrapperInstance.state.selectedParts[0].invalidQuantity).toBe(undefined);
    expect(wrapperInstance.state.selectedParts[1].invalidQuantity).toBe(undefined);
    expect(mockHandleUpdateReceivePartOrder).toBeCalledWith(updatedPayLoad);
  });

  it('should highlight receiving text box when saving with no quantity', () => {
    const mockHandleUpdateReceivePartOrder = jest.fn();

    const mockPart = {
      partNbr: '310-0103',
      brandNbr: 1001,
      brandName: 'MAKITA',
      partLocation: '',
      quantityOrdered: 10,
      partDescription: 'GASKET, TRANSMISSION-2',
      partsOrderDetailId: 2,
      previousQuantityReceived: 1,
      currentQuantityReceived: 0
    };

    const mockPart1Ar = [mockPart];

    const wrapperInstance = wrapper.instance();
    wrapperInstance.handleUpdateReceivePartOrder = mockHandleUpdateReceivePartOrder;
    wrapper.instance().forceUpdate();
    wrapperInstance.state.selectedParts = mockPart1Ar;
    wrapperInstance.state.selectedPurchaseOrderNbr = '971072952';
    wrapperInstance.handleSaveReceivePart();
    expect(wrapperInstance.state.selectedParts[0].invalidQuantity).toBe(true);
  });

  it('should handle UpdateReceivePartOrder for showSuccessMessage', async () => {
    wrapper = shallow(<ReceiveOrders {...props} />);
    const wrapperInstance = wrapper.instance();
    const spy = jest.spyOn(props.actions, 'updateReceivePartOrder');
    const showSuccessMessageSpy = jest.spyOn(props.actions, 'showSuccessMessage');
    wrapper.setProps({ updatePartOrdersError: false });
    await wrapperInstance.handleUpdateReceivePartOrder(payLoad);
    expect(spy).toBeCalledWith(payLoad);
    expect(showSuccessMessageSpy).toBeCalledWith('Parts Received: The parts have been received in to your inventory');
    spy.mockReset();
    spy.mockRestore();
    showSuccessMessageSpy.mockReset();
    showSuccessMessageSpy.mockRestore();
  });

  it('should handle UpdateReceivePartOrder for showMessage', async () => {
    const wrapperInstance = wrapper.instance();
    const showMessageSpy = jest.spyOn(props.actions, 'showMessage');
    wrapper.setProps({ updatePartOrdersError: true });
    await wrapperInstance.handleUpdateReceivePartOrder(payLoad);
    expect(showMessageSpy).toBeCalledWith('System Failure: Unable to receive parts, please try again later');
    showMessageSpy.mockReset();
    showMessageSpy.mockRestore();
  });

  it('should call getPurchaseOrders action on componentWillMount', () => {
    const spy = jest.spyOn(props.actions, 'getPurchaseOrders');
    const wrapperInstance = wrapper.instance();
    wrapperInstance.componentWillMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render properly with selectedParts in state ', () => {
    const mockPart = {
      partNbr: '310-0103',
      brandNbr: 1001,
      brandName: 'MAKITA',
      partLocation: '',
      quantityOrdered: 10,
      partDescription: 'GASKET, TRANSMISSION-2',
      partsOrderDetailId: 2,
      previousQuantityReceived: 1,
      currentQuantityReceived: 1
    };

    const inventoryWithNoLocation = {
      allocatedQuantity: 0,
      availableQuantity: 10,
      brandName: 'Eddy Floor Scraper',
      brandNbr: 1028,
      location: '',
      partDescription: 'GASKET, TRANSMISSION',
      partNbr: '310-0101',
      storeNbr: '9710'
    };

    const mockSelectedParts = [mockPart];
    wrapper.setState({ selectedParts: mockSelectedParts });
    wrapper.setProps({ inventory: [inventoryWithNoLocation] });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
