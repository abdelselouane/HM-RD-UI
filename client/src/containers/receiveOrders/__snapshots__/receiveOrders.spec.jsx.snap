// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReceiveOrders /> should render properly 1`] = `
<Fragment>
  <Table
    className=""
    clearOnHover={false}
    columns={
      Array [
        Object {
          "key": "orderCreatedDate",
          "name": "Date Created",
          "orderBy": Object {
            "className": "",
            "key": "orderCreatedDate",
          },
          "sortable": true,
        },
        Object {
          "key": "purchaseOrderNbr",
          "name": "Purchase Order #",
          "orderBy": Object {
            "className": "",
            "key": "purchaseOrderNbr",
          },
          "sortable": true,
        },
        Object {
          "key": "workOrderNbr",
          "name": "Work order #",
          "orderBy": Object {
            "className": "",
            "key": "workOrderNbr",
          },
          "sortable": true,
        },
        Object {
          "key": "purchaseOrderStatDesc",
          "name": "Status",
          "orderBy": Object {
            "className": "",
            "key": "purchaseOrderStatDesc",
          },
          "sortable": true,
        },
        Object {
          "key": "receive",
          "name": "Receive",
        },
      ]
    }
    handleRowClick={[Function]}
    handleSortClick={[Function]}
    isAlternating={false}
    isExpandable={true}
    rows={
      Array [
        Object {
          "key": "971072952",
          "locationNbr": "9710",
          "orderCreatedDate": "2018-06-04",
          "partsDetails": Array [
            Object {
              "brandName": "MAKITA",
              "brandNbr": 1001,
              "currentQuantityReceived": 0,
              "partDescription": "GASKET, TRANSMISSION",
              "partLocation": "",
              "partNbr": "310-0101",
              "partsOrderDetailId": 1,
              "previousQuantityReceived": 1,
              "quantityOrdered": 1,
            },
            Object {
              "brandName": "MAKITA",
              "brandNbr": 1001,
              "currentQuantityReceived": 0,
              "partDescription": "GASKET, TRANSMISSION-2",
              "partLocation": "",
              "partNbr": "310-0102",
              "partsOrderDetailId": 2,
              "previousQuantityReceived": 1,
              "quantityOrdered": 10,
            },
          ],
          "partsOrderId": 1,
          "purchaseOrderNbr": "971072952",
          "purchaseOrderStatDesc": "SUBMITTED",
          "receive": <button
            className="button md"
            id="0-971072952"
            onClick={[Function]}
            style={
              Object {
                "margin": "10px 0px",
              }
            }
          >
            Receive Order
          </button>,
          "workOrderNbr": "mock#",
        },
      ]
    }
    showBody={true}
    showCheckbox={false}
    showHeader={true}
    showResults={true}
  />
</Fragment>
`;

exports[`<ReceiveOrders /> should render properly with selectedParts in state  1`] = `
<Fragment>
  <Table
    className=""
    clearOnHover={false}
    columns={
      Array [
        Object {
          "key": "orderCreatedDate",
          "name": "Date Created",
          "orderBy": Object {
            "className": "",
            "key": "orderCreatedDate",
          },
          "sortable": true,
        },
        Object {
          "key": "purchaseOrderNbr",
          "name": "Purchase Order #",
          "orderBy": Object {
            "className": "",
            "key": "purchaseOrderNbr",
          },
          "sortable": true,
        },
        Object {
          "key": "workOrderNbr",
          "name": "Work order #",
          "orderBy": Object {
            "className": "",
            "key": "workOrderNbr",
          },
          "sortable": true,
        },
        Object {
          "key": "purchaseOrderStatDesc",
          "name": "Status",
          "orderBy": Object {
            "className": "",
            "key": "purchaseOrderStatDesc",
          },
          "sortable": true,
        },
        Object {
          "key": "receive",
          "name": "Receive",
        },
      ]
    }
    handleRowClick={[Function]}
    handleSortClick={[Function]}
    isAlternating={false}
    isExpandable={true}
    rows={
      Array [
        Object {
          "key": "971072952",
          "locationNbr": "9710",
          "orderCreatedDate": "2018-06-04",
          "partsDetails": Array [
            Object {
              "brandName": "MAKITA",
              "brandNbr": 1001,
              "currentQuantityReceived": 0,
              "partDescription": "GASKET, TRANSMISSION",
              "partLocation": "",
              "partNbr": "310-0101",
              "partsOrderDetailId": 1,
              "previousQuantityReceived": 1,
              "quantityOrdered": 1,
            },
            Object {
              "brandName": "MAKITA",
              "brandNbr": 1001,
              "currentQuantityReceived": 0,
              "partDescription": "GASKET, TRANSMISSION-2",
              "partLocation": "",
              "partNbr": "310-0102",
              "partsOrderDetailId": 2,
              "previousQuantityReceived": 1,
              "quantityOrdered": 10,
            },
          ],
          "partsOrderId": 1,
          "purchaseOrderNbr": "971072952",
          "purchaseOrderStatDesc": "SUBMITTED",
          "receive": <button
            className="button md"
            id="0-971072952"
            onClick={[Function]}
            style={
              Object {
                "margin": "10px 0px",
              }
            }
          >
            Receive Order
          </button>,
          "workOrderNbr": "mock#",
        },
      ]
    }
    showBody={true}
    showCheckbox={false}
    showHeader={true}
    showResults={true}
  />
</Fragment>
`;
